app/queues/notifications/expiration-notification.server.ts

```diff
@@ -0,0 +1,160 @@
+      'arguments: type and id for updateNotificationStatus cannot be null or undefined'
+    );
+  }
+  switch (type) {
+    case RecordType.Certificate:
+      return prisma.certificate.update({
+        where: {
+          id,
+        },
+        data: {
+          lastNotified: new Date(),
+        },
+      });
+    case RecordType.DnsRecord:
+      return prisma.record.update({
+        where: {
+          id,
+        },
+        data: {
+          lastNotified: new Date(),
+        },
+      });
+  }
+};
+
+// fetch records with by status
+const getRecordsByExpiration = (type: RecordType) => {
+  let monthFromToday = new Date();
+  monthFromToday.setMonth(monthFromToday.getMonth() + 1);
+  if (!type) {
+    throw new Error('argument: type for getRecordsByExpiration cannot be null or undefined');
+  }
+  switch (type) {
+    case RecordType.Certificate:
+      return prisma.certificate.findMany({
+        where: {
+          validTo: {
+            lte: monthFromToday,
+          },
+          lastNotified: null,
+        },
+        select: {
+          user: true,
+          id: true,
+          lastNotified: true,
+        },
+      });
+    case RecordType.DnsRecord:
+      return prisma.record.findMany({
+        where: {
+          expiresAt: {
+            lte: monthFromToday,
+          },
+          lastNotified: null,
+        },
+        select: {
+          user: true,
+          id: true,
+          lastNotified: true,
+        },
+      });
+  }
+};
+// Queue initialization
+export const expirationNotificationQueue = new Queue<ExpirationStatusPayload>(
+  expirationNotificationQueueName,
+  {
+    connection: redis,
+  }
+);
+
+expirationNotificationQueue.on('error', (err) => {
+  logger.warn('Error running check for DNS Records/Certificates about to expire', err);
+});
+
+// function to add jobs
+export const addExpirationNotification = async (type: RecordType) => {
+  if (!type) {
+    throw new Error('argument: type for addExpirationCheckJobs cannot be null or undefined');
+  }
+  let jobName = `${expirationNotificationQueueName}-${type}`;
+  return expirationNotificationQueue.add(
+    jobName,
+    { type },
+    {
+      repeat: { every: 5 * 60 * 1000 },
+    }
+  );
+};
+// function to update notification and add notification jobs
+const updateStatusAndNotify = async (type: RecordType, id: number, data: NotificationData) => {
+  const { emailAddress, subject, message } = data;
+  await updateNotificationStatus(type, id);
+  await addNotification({
+    emailAddress,
+    subject,
+    message,
+  });
+};
+// worker instance to process DNS Record/Certificates expiration notification jobs
+const expirationNotificationWorker = new Worker<ExpirationStatusPayload>(
+  expirationNotificationQueueName,
+  async (job) => {
+    const { type } = job.data;
+    try {
+      logger.debug(`Processing job: ${job.name}`);
+      let records = await getRecordsByExpiration(type);
+      records.map(async (record) => {
```

Comment:
You are mixing synchronous and asynchronous code paths here.  Your `.map()` will complete immediately, but the array will contain a bunch of Promises that need to get awaited.  You can fix it like so:

```ts
await Promise.all(records.map(async (record) => {
  ...
  await updateStatusAndNotify(...);
}))
```